packages/jaspr_lints/lib/src/assists/component_assist.dart

```diff
@@ -0,0 +1,187 @@
+              '  const ');
+
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write('({super.key});\n\n  @override\n  State createState() => ');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write('State();\n}\n\nclass ');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write('State extends State<');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write('> {\n\n  @override\n  Iterable<Component> build(BuildContext context) sync* {\n'
+              '    yield ');
+          edit.addSimpleLinkedEdit('child', "div([])");
+          edit.write(';\n  }\n}\n');
+        });
+        if (!hasJasprImport) {
+          builder.importLibrary(Uri.parse('package:jaspr/jaspr.dart'));
+        }
+      });
+
+      reporter.createChangeBuilder(priority: 3, message: 'Create InheritedComponent').addDartFileEdit((builder) {
+        if (!hasJasprImport) {
+          builder.importLibrary(Uri.parse('package:jaspr/jaspr.dart'));
+        }
+
+        builder.addInsertion(target.end == 0 ? 1 : target.end, (edit) {
+          edit.write('class ');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write(' extends InheritedComponent {\n'
+              '  const ');
+
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write('({super.child, super.children, super.key});\n\n  static ');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write(' of(BuildContext context) {\n    final ');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write('? result = context.dependOnInheritedComponentOfExactType<');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write('>();\n    assert(result != null, \'No ');
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write(
+              ' found in context.\');\n    return result!;\n  }\n\n  @override\n  bool updateShouldNotify(covariant ');
+
+          edit.addSimpleLinkedEdit('name', nameSuggestion);
+          edit.write(' oldComponent) {\n    return false;\n  }\n}\n');
+        });
+      });
+    });
+
+    context.registry.addClassDeclaration((node) {
+      if (target.offset < node.offset || target.end > node.leftBracket.end) {
+        return;
+      }
+      if (node.extendsClause?.superclass.name2.lexeme != 'StatelessComponent') {
+        return;
+      }
+
+      MethodDeclaration? buildMethod;
+      List<String> members = [];
+      for (var m in node.members) {
+        if (m is MethodDeclaration && m.name.lexeme == "build") {
+          buildMethod = m;
+        } else if (m is FieldDeclaration) {
+          members.addAll(m.fields.variables.map((v) => v.name.lexeme));
+        }
+      }
+
+      reporter.createChangeBuilder(priority: 1, message: 'Convert to StatefulComponent').addDartFileEdit((builder) {
```

Comment:
When I see this code, I think that this big `run` method could be splitted into more simple methods named:

createStatelessComponent
createStatefulComponent
createInheritedComponent
createChangeBuilder
convertToAsyncStatelessComponent
etc..

Where every single method could look like:
```
createStatelessComponent(reporter) {
  reporter.createChangeBuilder(....);
  return reporter;
}
```

Reply:
Done